src/battle_util.c

```diff
@@ -5583,35 +5583,31 @@ u32 AbilityBattleEffects(u32 caseID, u32 battler, u32 ability, u32 special, u32
             if (!(gMoveResultFlags & MOVE_RESULT_NO_EFFECT)
              && !gProtectStructs[gBattlerAttacker].confusionSelfDmg
              && TARGET_TURN_DAMAGED
-             && IsBattlerAlive(battler))
+             && IsBattlerAlive(battler)
+             && gBattleMons[gBattlerTarget].species != SPECIES_CRAMORANT)
             {
-                // TODO: Convert this to a proper FORM_CHANGE type.
-                if (gBattleMons[gBattlerTarget].species == SPECIES_CRAMORANT_GORGING)
-                {
-                    gBattleMons[gBattlerTarget].species = SPECIES_CRAMORANT;
                     if (GetBattlerAbility(gBattlerAttacker) != ABILITY_MAGIC_GUARD)
                     {
                         gBattleMoveDamage = GetNonDynamaxMaxHP(gBattlerAttacker) / 4;
                         if (gBattleMoveDamage == 0)
                             gBattleMoveDamage = 1;
                     }
+                  
                     BattleScriptPushCursor();
-                    gBattlescriptCurrInstr = BattleScript_GulpMissileGorging;
-                    effect++;
-                }
-                else if (gBattleMons[gBattlerTarget].species == SPECIES_CRAMORANT_GULPING)
-                {
-                    gBattleMons[gBattlerTarget].species = SPECIES_CRAMORANT;
-                    if (GetBattlerAbility(gBattlerAttacker) != ABILITY_MAGIC_GUARD)
+                    
+                    switch(gBattleMons[gBattlerTarget].species)
                     {
-                        gBattleMoveDamage = GetNonDynamaxMaxHP(gBattlerAttacker) / 4;
-                        if (gBattleMoveDamage == 0)
-                            gBattleMoveDamage = 1;
+                        case SPECIES_CRAMORANT_GORGING:             
+                            TryBattleFormChange(battler, FORM_CHANGE_HIT_BY_MOVE);                                        
+                            gBattlescriptCurrInstr = BattleScript_GulpMissileGorging; 
+                            break;
+                        case SPECIES_CRAMORANT_GULPING: 
+                        default:           
+                            TryBattleFormChange(battler, FORM_CHANGE_HIT_BY_MOVE);                                            
+                            gBattlescriptCurrInstr = BattleScript_GulpMissileGulping; 
+                            break;
                     }
-                    BattleScriptPushCursor();
-                    gBattlescriptCurrInstr = BattleScript_GulpMissileGulping;
-                    effect++;
-                }
+                    effect++;                
```

Comment:
Can you move `effect++` to both switch cases,  `SPECIES_CRAMORANT_GORGING` and `SPECIES_CRAMORANT_GULPING`? The effect should only be increased if one of Cramorant species forms is correct otherwise it would be a bug.